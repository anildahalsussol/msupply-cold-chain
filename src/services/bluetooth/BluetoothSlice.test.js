import {
  BluetoothStateReducer,
  BluetoothStateActions,
  PassiveBluetoothReducer,
  PassiveBluetoothActions,
} from './bluetoothSlice';
import { TEMPERATURE_SYNC_STATE , BLUETOOTH_SERVICE, PASSIVE_TEMPERATURE_SYNC_STATE } from '~constants';


describe('BluetoothStateActions', () => {
  it('downloadTemperatures', () => {
    expect(BluetoothStateActions.downloadTemperatures()).toMatchSnapshot();
  });
  it('startTemperatureSync', () => {
    expect(BluetoothStateActions.startTemperatureSync()).toMatchSnapshot();
  });
  it('scanForSensors', () => {
    expect(BluetoothStateActions.scanForSensors()).toMatchSnapshot();
  });
  it('complete', () => {
    expect(BluetoothStateActions.complete()).toMatchSnapshot();
  });
  it('start', () => {
    expect(BluetoothStateActions.start()).toMatchSnapshot();
  });
  it('disable', () => {
    expect(BluetoothStateActions.disable()).toMatchSnapshot();
  });
  it('error', () => {
    expect(BluetoothStateActions.error()).toMatchSnapshot();
  });
});

describe('PassiveBluetoothActions', () => {
  it('startTimer', () => {
    expect(PassiveBluetoothActions.startTimer()).toMatchSnapshot();
  });
  it('decrementTimer', () => {
    expect(PassiveBluetoothActions.decrementTimer()).toMatchSnapshot();
  });
  it('completeTimer', () => {
    expect(PassiveBluetoothActions.completeTimer()).toMatchSnapshot();
  });
  it('start', () => {
    expect(PassiveBluetoothActions.start()).toMatchSnapshot();
  });
  it('stop', () => {
    expect(PassiveBluetoothActions.stop()).toMatchSnapshot();
  });
  it('stopped', () => {
    expect(PassiveBluetoothActions.stopped()).toMatchSnapshot();
  });
  it('complete', () => {
    expect(PassiveBluetoothActions.complete()).toMatchSnapshot();
  });
});

describe('BluetoothStateReducer', () => {
  it('downloadTemperatures', () => {
    const initialState = { state: TEMPERATURE_SYNC_STATE.ENABLED };
    expect(
      BluetoothStateReducer(initialState, BluetoothStateActions.downloadTemperatures())
    ).toEqual(initialState);
  });
  it('startTemperatureSync', () => {
    const initialState = { state: TEMPERATURE_SYNC_STATE.ENABLED };
    expect(
      BluetoothStateReducer(initialState, BluetoothStateActions.startTemperatureSync())
    ).toEqual(initialState);
  });
  it('complete', () => {
    const initialState = { state: TEMPERATURE_SYNC_STATE.IN_PROGRESS };
    expect(BluetoothStateReducer(initialState, BluetoothStateActions.complete())).toEqual({
      state: TEMPERATURE_SYNC_STATE.ENABLED,
    });
  });
  it('start', () => {
    const initialState = { state: TEMPERATURE_SYNC_STATE.ENABLED };
    expect(BluetoothStateReducer(initialState, BluetoothStateActions.start())).toEqual({
      state: TEMPERATURE_SYNC_STATE.IN_PROGRESS,
    });
  });
  it('disable', () => {
    const initialState = { state: TEMPERATURE_SYNC_STATE.ENABLED };
    expect(BluetoothStateReducer(initialState, BluetoothStateActions.disable())).toEqual({
      state: TEMPERATURE_SYNC_STATE.DISABLED,
    });
  });
  it('error', () => {
    const initialState = { state: TEMPERATURE_SYNC_STATE.ENABLED };
    expect(BluetoothStateReducer(initialState, BluetoothStateActions.error())).toEqual({
      state: TEMPERATURE_SYNC_STATE.ERROR,
    });
  });
});

describe('PassiveBluetoothReducer', () => {
  it('startTimer', () => {
    const initialState = { timer: null, state: null };
    expect(PassiveBluetoothReducer(initialState, PassiveBluetoothActions.startTimer())).toEqual({
      timer: BLUETOOTH_SERVICE.DEFAULT_PASSIVE_DOWNLOAD_DELAY,
      state: null,
    });
  });
  it('decrementTimer', () => {
    const initialState = { timer: BLUETOOTH_SERVICE.DEFAULT_PASSIVE_DOWNLOAD_DELAY, state: null };
    expect(PassiveBluetoothReducer(initialState, PassiveBluetoothActions.decrementTimer())).toEqual(
      {
        timer:
          BLUETOOTH_SERVICE.DEFAULT_PASSIVE_DOWNLOAD_DELAY - BLUETOOTH_SERVICE.DEFAULT_TIMER_DELAY,
        state: null,
      }
    );
  });
  it('completeTimer', () => {
    const initialState = { timer: BLUETOOTH_SERVICE.DEFAULT_PASSIVE_DOWNLOAD_DELAY, state: null };
    expect(PassiveBluetoothReducer(initialState, PassiveBluetoothActions.completeTimer())).toEqual({
      timer: null,
      state: null,
    });
  });
  it('start', () => {
    const initialState = { timer: null, state: null };
    expect(PassiveBluetoothReducer(initialState, PassiveBluetoothActions.start())).toEqual({
      timer: null,
      state: PASSIVE_TEMPERATURE_SYNC_STATE.IN_PROGRESS,
    });
  });
  it('stop', () => {
    const initialState = { timer: null, state: null };
    expect(PassiveBluetoothReducer(initialState, PassiveBluetoothActions.stop())).toEqual({
      timer: null,
      state: null,
    });
  });
  it('stopped', () => {
    const initialState = { timer: null, state: null };
    expect(PassiveBluetoothReducer(initialState, PassiveBluetoothActions.stopped())).toEqual({
      timer: null,
      state: PASSIVE_TEMPERATURE_SYNC_STATE.STOPPED,
    });
  });
  it('complete', () => {
    const initialState = { timer: null, state: null };
    expect(PassiveBluetoothReducer(initialState, PassiveBluetoothActions.complete())).toEqual({
      timer: null,
      state: PASSIVE_TEMPERATURE_SYNC_STATE.WAITING,
    });
  });
});
