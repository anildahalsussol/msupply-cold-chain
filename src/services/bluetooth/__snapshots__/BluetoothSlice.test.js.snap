// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BluetoothStateActions complete 1`] = `
Object {
  "payload": undefined,
  "type": "bluetooth_state/complete",
}
`;

exports[`BluetoothStateActions disable 1`] = `
Object {
  "payload": undefined,
  "type": "bluetooth_state/disable",
}
`;

exports[`BluetoothStateActions downloadTemperatures 1`] = `
Object {
  "payload": undefined,
  "type": "bluetooth_state/downloadTemperatures",
}
`;

exports[`BluetoothStateActions error 1`] = `
Object {
  "payload": undefined,
  "type": "bluetooth_state/error",
}
`;

exports[`BluetoothStateActions scanForSensors 1`] = `
Object {
  "payload": undefined,
  "type": "bluetooth_state/scanForSensors",
}
`;

exports[`BluetoothStateActions start 1`] = `
Object {
  "payload": undefined,
  "type": "bluetooth_state/start",
}
`;

exports[`BluetoothStateActions startTemperatureSync 1`] = `
Object {
  "payload": undefined,
  "type": "bluetooth_state/startTemperatureSync",
}
`;

exports[`PassiveBluetoothActions complete 1`] = `
Object {
  "payload": undefined,
  "type": "passive_bluetooth/complete",
}
`;

exports[`PassiveBluetoothActions completeTimer 1`] = `
Object {
  "payload": undefined,
  "type": "passive_bluetooth/completeTimer",
}
`;

exports[`PassiveBluetoothActions decrementTimer 1`] = `
Object {
  "payload": undefined,
  "type": "passive_bluetooth/decrementTimer",
}
`;

exports[`PassiveBluetoothActions start 1`] = `
Object {
  "payload": undefined,
  "type": "passive_bluetooth/start",
}
`;

exports[`PassiveBluetoothActions startTimer 1`] = `
Object {
  "payload": undefined,
  "type": "passive_bluetooth/startTimer",
}
`;

exports[`PassiveBluetoothActions stop 1`] = `
Object {
  "payload": undefined,
  "type": "passive_bluetooth/stop",
}
`;

exports[`PassiveBluetoothActions stopped 1`] = `
Object {
  "payload": undefined,
  "type": "passive_bluetooth/stopped",
}
`;
